odoo/addons/base/models/res_users.py

```diff
@@ -643,6 +643,10 @@ def _login(cls, db, login, password):
                         raise AccessDenied()
                     user = user.with_user(user)
                     user._check_credentials(password)
+                    tz = request.httprequest.cookies.get('tz') if request else None
```

Comment:
good point: we need to be careful that `_login` might called without a `request`